### DISCUSSION TITLE
Group commands into panes similar to warp terminal

### DISCUSSION THREAD

**Discussion by lkuoch** [zellij-org/zellij#3088] [Ideas]
Just want to say zellij is so amazing and I use it everyday with so much joy @imsnif!

There's this amazing feature in warp terminal (https://www.warp.dev/) where executed commands are grouped into "mini-panes" where they can be navigated over and have their own scroll buffer.

Once you try it there's no going back however it's only available on macos and lives as a terminal app.

With the solid foundations of with panes and tabs built into zellij, having a option of a command pane with the features described above would elevate the experience to the next level.

**Comment by imsnif** [comment]
I'm pretty sure we were the inspiration for this feature :)

Check out command panes in Zellij, you can run them (for example) with `zellij run -- cargo test` or `zr cargo test` if you use the provided alias. You can also run them as floating panes with `zrf` or as in-place panes with `zri`. They will show you the exit status and allow you to re-run the command with `ENTER`. Enjoy.

**Reply by lkuoch** [reply]
I was experimenting with zellij run but I don't think it's quite there yet but could be extended to support something similar to the warp terminal behavior.
(Could be that every terminal command is a zellij run but it creates a inline pane)

Having commands run in the root pane but it's nested block for each new command where you can navigate between is what I'm looking for.


Something like this

```
---ROOT PANE------------
| [Command 1]          | <- separate pane, own history, re-runable, resizeable
| ----------------------
| [Command 2]          |
| ----------------------
| [Command 3]          |
|-----------------------
```

**Comment by giyany** [comment]
Hi,

I wrote a short bash script to do what you're looking for and named it 'marathon'.
You'll want to download preexec 

```
curl https://raw.githubusercontent.com/rcaloras/bash-preexec/master/bash-preexec.sh -o ~/.bash-preexec.sh
```
then save and source this code
```bash
. bash-preexec.sh
shopt -s extdebug
preexec() { zellij run -- $1; return 1; }
```

I might turn this into a repository later and improve usability. It's great that you enjoy Warp - there are many wonderful tools out there and some have inspired Zellij. In any case, this solution isn't perfect (yet) but has the advantage of not spying nor collecting your data.

**Reply by lkuoch** [reply]
@giyany great work! It's a bit wonky on my end but that's probably because of my custom layout
@imsnif above is a great poc of what could be supported natively I believe
